### DISCUSSION TITLE
Group all update types

### DISCUSSION THREAD

**Discussion by tpansino** [renovatebot/renovate#13355] [Request Help]
I'm using the hosted app, trying to combine all packages AND update types together into one PR that I can review on Monday mornings. Here's my config:

```
{
  "extends": ["config:base"],
  "lockFileMaintenance": {
    "enabled": true
  },
  "packageRules": [
    {
      "matchUpdateTypes": ".*",
      "matchPackagePatterns": ".*",
      "groupName": "all dependencies",
      "groupSlug": "all"
    }
  ],
  "schedule": ["before 3am on Monday"],
  "separateMajorMinor": true,
  "timezone": "America/Los_Angeles"
}
```

The problem is this doesn't seem to affect `lockFileMaintenance`, as I received a lock file maintenance PR this morning. What am I doing wrong here?

Similar to #13284.

**Comment by viceice** [comment]
Remove `matchUpdateTypes`, it doesn't support wildcard. The rules will apply to all update types by default.

**Reply by rarkins** [reply]
Seems that they want lockFileMaintenance in the same PR too. I know that can work although it wasn't our original intention for it

**Reply by tpansino** [reply]
> it doesn't support wildcard

Hmm, that was not clear to me from the documentation. So is `matchPackagePatterns` the only `match` that supports wildcard/regex?

> Seems that they want lockFileMaintenance in the same PR too. I know that can work although it wasn't our original intention for it

Yes, that is correct. What I would like is to receive a PR every Monday, containing all lock file updates and minor/patch updates. Is there a problem with grouping updates this way?

Edit: I have created a [minimal repository](https://github.com/tpansino/renovatebot-test/pulls) which I think should be doing what I want, but observe that it has created 2 PRs - one for lock file updates and one for a patch update.

**Reply by rarkins** [reply]
> > it doesn't support wildcard
> 
> Hmm, that was not clear to me from the documentation. So is `matchPackagePatterns` the only `match` that supports wildcard/regex?

Fields in `packageRules` are individually documented with what they support. Read https://docs.renovatebot.com/configuration-options/#packagerules on down.

If you think documentation can be improved, PRs are welcome.

> Yes, that is correct. What I would like is to receive a PR every Monday, containing all lock file updates and minor/patch updates. Is there a problem with grouping updates this way?

Yes, the problem is that lock file maintenance is intended to be a separate PR

**Reply by tpansino** [reply]
> Read https://docs.renovatebot.com/configuration-options/#packagerules on down.

The `subType` and `format` rows in the tables is what I was missing, thank you.

> Yes, the problem is that lock file maintenance is intended to be a separate PR

Thank you, but I'm not clear: why is that the intended usage? The package rules docs seem to suggest a user can group any type of update, why are lock file updates fundamentally different than a minor/patch update?

**Reply by rarkins** [reply]
You're mixing up the concepts of "lock file update" with "lock file maintenance". The latter is fundamentally different because it doesn't update any specific dependency and instead does a full refresh/regeneration of the lock file. With npm for example it will *delete* the lock file before running `npm install` to regenerate it.

**Reply by tpansino** [reply]
> You're mixing up the concepts of "lock file update" with "lock file maintenance".

Hmm, ok. Maybe I should step back and describe my goal?

Every Monday, I would like to receive a single PR which has all dependencies updated to latest versions. So any minor/patch updates to `.python-version`, plus updates to locked dependencies in the `poetry.lock` file. Basically, I want the 2 PRs currently showing on my [minimal repo](https://github.com/tpansino/renovatebot-test/pulls), but in 1 PR.

Does lock file maintenance _have_ to be a separate PR for the tool to function? It seems like these update types should be combinable, but if you say they are not, I believe you. I just wish to understand why. 🙂

**Reply by rarkins** [reply]
It's mostly a technical limitation. You could try the following to see if it works:

```
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "lockFileMaintenance"],
      "matchPackagePatterns": [".*"],
      "groupName": "all dependencies",
      "groupSlug": "all"
    }
  ],
```

I strongly recommend you don't add `major` update types to that list, because you'll pretty soon get at least one breaking change which then stops you from merging the PR, and therefore updating *anything at all*.

**Reply by tpansino** [reply]
It looks like it does create only 1 [PR](https://github.com/tpansino/renovatebot-test2/pull/1), but it doesn't contain the lock file updates. I guess that's the technical limitation 😄 

![image](https://user-images.githubusercontent.com/2768420/148810401-62796055-a39b-4841-85d2-04f2ef5a9c87.png)

I will file a Feature Request, thank you for your time.

**Reply by rarkins** [reply]
What did the logs say?

**Comment by viceice** [comment]
`separateMajorMinor` needs to be false to get all updates into the group.

We recommend to not group all  major updates!
